bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/internal/BlueZBluetoothDevice.java

```diff
@@ -469,7 +469,7 @@ private void convertCharacteristicProperties(BluetoothGattCharacteristic dBusBlu
     @Override
     public CompletableFuture<byte[]> readCharacteristic(BluetoothCharacteristic characteristic) {
         BluetoothDevice dev = device;
-        if (dev == null || !dev.isConnected()) {
+        if (dev == null || dev.isConnected() == null || !dev.isConnected()) {
```

Comment:
Looking again, maybe it could be simplified:
```suggestion
        if (!Boolean.TRUE.equals(dev.isConnected()) {
```

Reply:
Still need to check `dev == null` so it would be:
`if (dev == null || !Boolean.TRUE.equals(dev.isConnected()) {`

In that case i prefer the more clear variant as it is now.

Reply:
Right, of course! Oh, I miss null-conditional operators:
```suggestion
        if (!Boolean.TRUE.equals(dev?.isConnected()) {
```

🙂

Even with that null check still needed, would you still consider it, taking into account that.
- You would call the getter only once.
- It would be more obvious that `dev.isConnected()` is `Boolean`, rather than the now slightly confusing auto-unboxing after the second null check.
- It would therefore be less likely that someone removes the second null check again and reintroduces the bug - because it's not obvious that it's needed when reading the line.

Reply:
Totally agree on the null conditional operator. Really don't like Boolean, but that is another story ;-)
Changed it once again, please check if all is oke now.